src/infrastructure/figma/figma-service.ts

```diff
@@ -86,6 +86,53 @@ export class FigmaService {
 		}
 	};
 
+	fetchDesignsByIds = async (
+		designIds: FigmaDesignIdentifier[],
+		atlassianUserId: string,
+	): Promise<AtlassianDesign[]> => {
+		if (!designIds.length) {
+			return [];
+		}
+
+		// Ensure all design identifiers have the same file key
+		const fileKey = designIds[0].fileKey;
+		for (const designId of designIds.slice(1)) {
```

Comment:
**nitpick:** Consider using `Array.prototype.every`, which could help to simplify the code:

```ts
const fileKey = designIds[0].fileKey;
const haveSameFileKey = designIds.every((designId) => designId.fileKey === fileKey);

if (!haveSameFileKey) {
	throw new FigmaServiceError('designIds must all have the same fileKey');
}
```

Reply:
TIL `Array.prototype.every`